lib/puppet/pops/types/type_calculator.rb

```diff
@@ -1192,22 +1201,34 @@ def tuple_entry_at(tuple_t, from, to, index)
   # @api private
   #
   def assignable_PStructType(t, t2)
-    return true if t == t2 || t.elements.empty? && (t2.is_a?(Types::PHashType))
-    h = t.hashed_elements
+    els = t.elements
```

Comment:
'els' as short for elements ? (not such a great name, looks like misspelled 'else'